### PR TITLE
fix(batch): schedule single task for singleton table scan

### LINKED ISSUE #4164: batch: singleton seq scan on local mode
**Describe the bug**
Currently, when scanning from a singleton mview, we still schedule multiple scans.

**To Reproduce**
```sql
create table t (v int);
create materialized view mv as select v from t order by v limit 10;

explain verbose select * from mv;
                          QUERY PLAN                          
--------------------------------------------------------------
 BatchExchange { order: [], dist: [Single] }
   BatchScan { table: mv, columns: [mv.v], distribution: [] }
```

The distribution is `[]`, that is, `HashShard` with empty keys. This is incorrect, we should use `Single` instead.

https://github.com/singularity-data/risingwave/blob/eb8109fc122ad51a8ccfcf4fc1efc03d79bfdb0a/src/frontend/src/optimizer/plan_node/batch_seq_scan.rs#L75-L93

#4153 introduced a fix while commented out, since it won't work on local mode now. We should fix this.

**Expected behavior**
```
                          QUERY PLAN                          
---------------------------------------------------------------
 BatchScan { table: mv, columns: [mv.v], distribution: Single }
```

**Additional context**
#4153 fixes the similar issue of `StreamScan`.
#3269 is related.

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

For singleton table, we make the distribution of the table scan `Single` instead of `UpstreamHashShard(vec![])`.

Per discussion with @liurenjie1024, as the root stage is always executed on the frontend, we insert an additional `Single` if necessary to ensure the table scan runs on the compute node just like what we do for DMLs.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- Close #4164

### CODE REVIEW

**Review by xxchan — APPROVED** [review]
LGTM

**Review by liurenjie1024 — APPROVED** [review]
LGTM

File: src/frontend/planner_test/tests/testdata/batch_dist_agg.yaml [hunk 1]
```diff
@@ -160,9 +159,10 @@
...
+          └─BatchScan { table: s, columns: [s.k, s.v], distribution: Single }
   batch_local_plan: |
-    BatchProject { exprs: [max(s.v)] }
-    └─BatchHashAgg { group_key: [s.k], aggs: [max(s.v)] }
-      └─BatchExchange { order: [], dist: Single }
-        └─BatchScan { table: s, columns: [s.k, s.v], distribution: UpstreamHashShard() }
+    BatchExchange { order: [], dist: Single }
+    └─BatchProject { exprs: [max(s.v)] }
+      └─BatchHashAgg { group_key: [s.k], aggs: [max(s.v)] }
+        └─BatchScan { table: s, columns: [s.k, s.v], distribution: Single }
```
Comment (BowenXiao1999) [inline src/frontend/planner_test/tests/testdata/batch_dist_agg.yaml:168]: Take this for example, root fragment task is always executed on FE, while one is exchange singleton and another is hash agg. 

What's the benefit of doing so? 🤔

**Review by BowenXiao1999 — APPROVED** [review]

File: src/frontend/planner_test/tests/testdata/batch_dist_agg.yaml (hunk 1 above)
Comment (liurenjie1024) [reply src/frontend/planner_test/tests/testdata/batch_dist_agg.yaml:168]: It's for this bug: https://github.com/risingwavelabs/risingwave/issues/4164

**Pull request merged**